frontend/scripts/tests/puppeteer/profile-actor.test.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable no-console */
+import puppeteer from 'puppeteer';
+
+import mocks from '../mocks';
+
+let page;
+let browser;
+const baseUrl = 'http://0.0.0.0:8081';
+
+const openBrowser = visible =>
+  visible
+    ? {
+        headless: false,
+        slowMo: 80,
+        args: [`--window-size=1920,1080`]
+      }
+    : { args: ['--no-sandbox'] };
+
+beforeAll(async () => {
+  browser = await puppeteer.launch(openBrowser(false));
+  page = await browser.newPage();
+  await page.setRequestInterception(true);
+  page.on('request', interceptedRequest => {
+    const url = interceptedRequest
+      .url()
+      .replace('https:', '')
+      .replace('http:', '');
+
+    if (url in mocks) {
+      console.warn(`URL (${url}) intercepted and response mocked`);
+      setTimeout(
+        () =>
+          interceptedRequest.respond({
+            status: 200,
+            contentType: 'application/json',
+            headers: {
+              'Access-Control-Allow-Origin': '*'
+            },
+            body: JSON.stringify(mocks[url])
+          }),
+        300
+      );
+    } else {
+      interceptedRequest.continue();
+    }
+  });
+});
+
+afterAll(() => {
+  browser.close();
```

Comment:
this could also be moved to a helper like `helpers/destroy-app.js` or sth similar

Reply:
I'm not sure about this one, I think it's clear enough. Maybe we could refactor it into a `destroy();` method but we'd still need to call it in each test's `afterAll`. It's a trade-off.